### PR TITLE
Log clarification

### PR DESCRIPTION
**Pull request by mdaguete**
Added the ability to specify a fun for error and info logging. Also added the connection name to current logs to Identify connection in a system with multiple ones.

### CODE REVIEW

File: src/apns_connection.erl [hunk 1]
```diff
@@ -82,6 +93,9 @@ init(Connection) ->
                        , connection = Connection
                        , queue      = QID
                        , out_expires = Timeout
+                       , name = Name
+                       , error_logger_fun = Connection#apns_connection.error_logger_fun
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 97 is too long:                        , error_logger_fun = Connection#apns_connection.error_logger_fun.

File: src/apns_connection.erl [hunk 2]
```diff
@@ -239,8 +260,8 @@ handle_info( {ssl, SslSocket, Data}
       try call(Feedback, [{apns:timestamp(TimeT), bin_to_hexstr(Token)}])
       catch
         _:Error ->
-          error_logger:error_msg(
-            "Error trying to inform feedback token ~p:~n\t~p~n", [Token, Error])
+          ErrorLoggerFun(
+            "[ ~p ] Error trying to inform feedback token ~p: ~p", [Name, Token, Error])
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 264 is too long:             "[ ~p ] Error trying to inform feedback token ~p: ~p", [Name, Token, Error]).

File: src/apns_connection.erl [hunk 3]
```diff
@@ -336,14 +368,15 @@ do_build_payload([], Payload) ->
 
 -spec send_payload(tuple(), binary(), non_neg_integer(), binary(), binary(), integer()) ->
   ok | {error, term()}.
-send_payload(Socket, MsgId, Expiry, BinToken, Payload, Priority) ->
+send_payload(#state{out_socket = Socket, info_logger_fun = InfoLoggerFun, name = Name}
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 371 is too long: send_payload(#state{out_socket = Socket, info_logger_fun = InfoLoggerFun, name = Name}.

File: src/apns_connection.erl [hunk 4]
```diff
@@ -336,14 +368,15 @@ do_build_payload([], Payload) ->
 
 -spec send_payload(tuple(), binary(), non_neg_integer(), binary(), binary(), integer()) ->
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 369 is too long: -spec send_payload(tuple(), binary(), non_neg_integer(), binary(), binary(), integer()) ->.

File: src/apns_connection.erl [hunk 5]
```diff
@@ -42,7 +45,7 @@ stop(ConnId) ->
 -spec start_link(atom(), apns:connection()) ->
   {ok, pid()} | {error, {already_started, pid()}}.
 start_link(Name, Connection) ->
-  gen_server:start_link({local, Name}, ?MODULE, Connection, []).
+  gen_server:start_link({local, Name}, ?MODULE, [Name,Connection], []).
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 48

File: src/apns_connection.erl [hunk 6]
```diff
@@ -70,7 +73,15 @@ build_payload(Msg) ->
 
 %% @hidden
 -spec init(apns:connection()) -> {ok, state()} | {stop, term()}.
+init([Name,Connection]) ->
+  init(Name,Connection);
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 77

File: src/apns_connection.erl [hunk 7]
```diff
@@ -250,25 +271,36 @@ handle_info( {ssl, SslSocket, Data}
...
-    "Feedback server disconnected. Waiting ~p millis to connect again...~n",
-    [Connection#apns_connection.feedback_timeout]),
+handle_info({ssl_closed, SslSocket}
+           , State = #state{in_socket = SslSocket
+                           ,connection= Connection
+                           , info_logger_fun = InfoLoggerFun
+                           , name = Name
+                           }) ->
+  InfoLoggerFun(
+    "[ ~p ] Feedback server disconnected. Waiting ~p millis to connect again...",
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 281 is too long:     "[ ~p ] Feedback server disconnected. Waiting ~p millis to connect again...",.

File: src/apns_connection.erl [hunk 8]
```diff
@@ -82,6 +93,9 @@ init(Connection) ->
                        , connection = Connection
                        , queue      = QID
                        , out_expires = Timeout
+                       , name = Name
+                       , error_logger_fun = Connection#apns_connection.error_logger_fun
+                       , info_logger_fun  = Connection#apns_connection.info_logger_fun
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 98 is too long:                        , info_logger_fun  = Connection#apns_connection.info_logger_fun.

File: src/apns_connection.erl [hunk 9]
```diff
@@ -70,7 +73,15 @@ build_payload(Msg) ->
 
 %% @hidden
 -spec init(apns:connection()) -> {ok, state()} | {stop, term()}.
+init([Name,Connection]) ->
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 76

File: src/apns_connection.erl [hunk 10]
```diff
@@ -250,25 +271,36 @@ handle_info( {ssl, SslSocket, Data}
...
   end;
 
-handle_info({ssl_closed, SslSocket}, State = #state{in_socket = SslSocket,
-                                                    connection= Connection}) ->
-  error_logger:info_msg(
-    "Feedback server disconnected. Waiting ~p millis to connect again...~n",
-    [Connection#apns_connection.feedback_timeout]),
+handle_info({ssl_closed, SslSocket}
+           , State = #state{in_socket = SslSocket
+                           ,connection= Connection
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 276

File: src/apns_connection.erl [hunk 11]
```diff
@@ -151,9 +165,12 @@ handle_call(Request, _From, State) ->
...
+                             , connection = Connection
+                             , info_logger_fun = InfoLoggerFun
+                             , name = Name
+                             }) ->
   try
-    error_logger:info_msg("Reconnecting to APNS...~n"),
+    InfoLoggerFun("[ ~p ] Reconnecting to APNS...",[Name]),
     Timeout = epoch() + Connection#apns_connection.expires_conn,
     case open_out(Connection) of
       {ok, Socket} -> handle_cast(Msg, State#state{out_socket=Socket, out_expires = Timeout});
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 176 is too long:       {ok, Socket} -> handle_cast(Msg, State#state{out_socket=Socket, out_expires = Timeout});.

File: src/apns_connection.erl [hunk 12]
```diff
@@ -70,7 +73,15 @@ build_payload(Msg) ->
...
 -spec init(apns:connection()) -> {ok, state()} | {stop, term()}.
+init([Name,Connection]) ->
+  init(Name,Connection);
 init(Connection) ->
+  init(?MODULE,Connection).
+
+
+%% @hidden
+-spec init(atom(), apns:connection()) -> {ok, state() | {stop, term()}}.
+init(Name,Connection) ->
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 84

File: src/apns_connection.erl [hunk 13]
```diff
@@ -175,7 +192,7 @@ handle_cast(Msg, State) when is_record(Msg, apns_msg) ->
       Payload = build_payload(Msg),
       BinToken = hexstr_to_bin(Msg#apns_msg.device_token),
       apns_queue:in(State#state.queue, Msg),
-      case send_payload(Socket, Msg#apns_msg.id, Msg#apns_msg.expiry, BinToken, Payload, Msg#apns_msg.priority) of
+      case send_payload(State, Msg#apns_msg.id, Msg#apns_msg.expiry, BinToken, Payload, Msg#apns_msg.priority) of
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 195 is too long:       case send_payload(State, Msg#apns_msg.id, Msg#apns_msg.expiry, BinToken, Payload, Msg#apns_msg.priority) of.

File: src/apns_connection.erl [hunk 14]
```diff
@@ -70,7 +73,15 @@ build_payload(Msg) ->
 
 %% @hidden
 -spec init(apns:connection()) -> {ok, state()} | {stop, term()}.
+init([Name,Connection]) ->
+  init(Name,Connection);
 init(Connection) ->
+  init(?MODULE,Connection).
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 79

File: src/apns_connection.erl [hunk 15]
```diff
@@ -151,9 +167,12 @@ handle_call(Request, _From, State) ->
...
+                             , connection = Connection
+                             , info_logger_fun = InfoLoggerFun
+                             , name = Name
+                             }) ->
   try
-    error_logger:info_msg("Reconnecting to APNS...~n"),
+    InfoLoggerFun("[ ~p ] Reconnecting to APNS...",[Name]),
     Timeout = epoch() + Connection#apns_connection.expires_conn,
     case open_out(Connection) of
       {ok, Socket} -> handle_cast(Msg, State#state{out_socket=Socket, out_expires = Timeout});
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 178 is too long:       {ok, Socket} -> handle_cast(Msg, State#state{out_socket=Socket, out_expires = Timeout});.

File: src/apns_connection.erl [hunk 16]
```diff
@@ -175,7 +194,7 @@ handle_cast(Msg, State) when is_record(Msg, apns_msg) ->
       Payload = build_payload(Msg),
       BinToken = hexstr_to_bin(Msg#apns_msg.device_token),
       apns_queue:in(State#state.queue, Msg),
-      case send_payload(Socket, Msg#apns_msg.id, Msg#apns_msg.expiry, BinToken, Payload, Msg#apns_msg.priority) of
+      case send_payload(State, Msg#apns_msg.id, Msg#apns_msg.expiry, BinToken, Payload, Msg#apns_msg.priority) of
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Line 197 is too long:       case send_payload(State, Msg#apns_msg.id, Msg#apns_msg.expiry, BinToken, Payload, Msg#apns_msg.priority) of.

File: src/apns_connection.erl [hunk 17]
```diff
@@ -250,25 +274,37 @@ handle_info( {ssl, SslSocket, Data}
...
   end;
 
-handle_info({ssl_closed, SslSocket}, State = #state{in_socket = SslSocket,
-                                                    connection= Connection}) ->
-  error_logger:info_msg(
-    "Feedback server disconnected. Waiting ~p millis to connect again...~n",
-    [Connection#apns_connection.feedback_timeout]),
+handle_info({ssl_closed, SslSocket}
+           , State = #state{in_socket = SslSocket
+                           ,connection= Connection
```
Comment (gadgetci) [inline src/apns_connection.erl]: According to **Elvis**:
> Missing space after "," on line 279

**Comment by elbrujohalcon** [comment]
Great PR, @mdaguete !! Thank you :)

**Pull request merged**